nx_parallel/interface.py

```diff
@@ -1,125 +1,114 @@
-from nx_parallel.algorithms.bipartite.redundancy import node_redundancy
-from nx_parallel.algorithms.centrality.betweenness import (
-    betweenness_centrality,
-    edge_betweenness_centrality,
-)
-from nx_parallel.algorithms.shortest_paths.generic import all_pairs_all_shortest_paths
-from nx_parallel.algorithms.shortest_paths.weighted import (
-    all_pairs_dijkstra,
-    all_pairs_dijkstra_path_length,
-    all_pairs_dijkstra_path,
-    all_pairs_bellman_ford_path_length,
-    all_pairs_bellman_ford_path,
-    johnson,
-)
-from nx_parallel.algorithms.shortest_paths.unweighted import (
-    all_pairs_shortest_path,
-    all_pairs_shortest_path_length,
-)
-from nx_parallel.algorithms.efficiency_measures import local_efficiency
-from nx_parallel.algorithms.isolate import number_of_isolates
-from nx_parallel.algorithms.tournament import (
-    is_reachable,
-    tournament_is_strongly_connected,
-)
-from nx_parallel.algorithms.vitality import closeness_vitality
-from nx_parallel.algorithms.approximation.connectivity import (
-    approximate_all_pairs_node_connectivity,
-)
-from nx_parallel.algorithms.connectivity import connectivity
-from nx_parallel.algorithms.cluster import square_clustering
 import networkx as nx
+from typing import Union
+from operator import attrgetter
+
+from nx_parallel import algorithms
 
 __all__ = ["BackendInterface", "ParallelGraph"]
 
+NX_GTYPES = Union[nx.Graph, nx.DiGraph, nx.MultiGraph, nx.MultiDiGraph]
+
+ALGORITHMS = [
+    # Bipartite
+    "node_redundancy",
+    # Isolates
+    "number_of_isolates",
+    # Vitality
+    "closeness_vitality",
+    # Tournament
+    "is_reachable",
+    "tournament_is_strongly_connected",
+    # Centrality
+    "betweenness_centrality",
+    "edge_betweenness_centrality",
+    # Efficiency
+    "local_efficiency",
+    # Shortest Paths : generic
+    "all_pairs_all_shortest_paths",
+    # Shortest Paths : weighted graphs
+    "all_pairs_dijkstra",
+    "all_pairs_dijkstra_path_length",
+    "all_pairs_dijkstra_path",
+    "all_pairs_bellman_ford_path_length",
+    "all_pairs_bellman_ford_path",
+    "johnson",
+    # Clustering
+    "square_clustering",
+    # Shortest Paths : unweighted graphs
+    "all_pairs_shortest_path",
+    "all_pairs_shortest_path_length",
+    # Approximation
+    "approximate_all_pairs_node_connectivity",
+    # Connectivity
+    "connectivity.all_pairs_node_connectivity",
+]
+
 
 class ParallelGraph:
-    """A wrapper class for networkx.Graph, networkx.DiGraph, networkx.MultiGraph,
-    and networkx.MultiDiGraph."""
+    """
```

Comment:
I agree that we need the doc_string final `"""` to be on its own line (otherwise the doc_string doesn't end with an EOL char). But I also think we should not have the starting `"""` on its own line. That adds a blank line at the beginning of the doc_string. And sphinx looks at the first line to find text that is used when creating a link to that doc_string. I guess we aren't using sphinx for docs in this library. But it seems better to be uniform with doc_string spacing.

I'm suggesting the first line of test appear with the starting `"""` and the final `"""` be on a line with no text. But this is a nitpicky choice so I'm fine if this goes in as is.

Reply:
Absolutely! Fixed.